### PR TITLE
Show changes

### PR DESCRIPTION
**Pull request by indranilatcal**
Here is a changeset with show changes. For now, done only for net45 for Documents. If this looks okay, can continue for other versions and apply to Entites.

### CODE REVIEW

**Comment by indranilatcal** [comment]
Was trying to apply this to Entities but there it cannot deserialze the response unless it comes out as json. Shows returning in any other format(html or xml) fails to work with JsonReader which gets used by default. Do we see much value with adding this to entities unless shows return jsons?

**Comment by danielwertheim** [comment]
No.

**Comment by indranilatcal** [comment]
In that case, I think shows appear to be at a logical point where you can have a look at the pattern applied to .NET 4.5. Once I have your feedback, I'd apply it to other projects.

**Comment by indranilatcal** [comment]
Did you get a chance to review this?

File: source/projects/MyCouch.Net45/Contexts/Documents.cs [hunk 1]
```diff
@@ -21,11 +22,14 @@ public class Documents : ApiContextBase<IDbConnection>, IDocuments
...
         protected PutDocumentHttpRequestFactory PutDocumentHttpRequestFactory { get; set; }
         protected DeleteDocumentHttpRequestFactory DeleteDocumentHttpRequestFactory { get; set; }
+        protected QueryShowHttpRequestFactory QueryShowHttpRequestFactory { get; set; }
         protected DocumentResponseFactory DocumentReponseFactory { get; set; }
         protected DocumentHeaderResponseFactory DocumentHeaderReponseFactory { get; set; }
         protected BulkResponseFactory BulkReponseFactory { get; set; }
+        protected RawResponseFactory RawResponseFactory { get; set; }
 
         public ISerializer Serializer { get; private set; }
+        public ISerializer DocumentSerializer { get; private set; }
```
Comment (danielwertheim) [inline source/projects/MyCouch.Net45/Contexts/Documents.cs]: This seems to be unused

File: source/projects/MyCouch.Net45/ShowIdentity.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        public ShowIdentity(string designDocument, string name)
+        {
+            Ensure.That(designDocument, "designDocument").IsNotNullOrWhiteSpace();
+            Ensure.That(name, "name").IsNotNullOrWhiteSpace();
+
+            DesignDocument = designDocument;
+            Name = name;
+        }
+
+        protected ShowIdentity(string name)
```
Comment (danielwertheim) [inline source/projects/MyCouch.Net45/ShowIdentity.cs]: This is not being used.

File: source/projects/MyCouch.Net45/Requests/QueryShowRequest.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+using MyCouch.Querying;
+using System;
+using System.Collections.Generic;
+
+namespace MyCouch.Requests
+{
+#if !PCL
+    [Serializable]
+#endif
+    public class QueryShowRequest : Request, IShowParameters
```
Comment (danielwertheim) [inline source/projects/MyCouch.Net45/Requests/QueryShowRequest.cs]: Would be awesome with API comments on the public properties.

File: source/projects/MyCouch.Net45/IShowParameters.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿
+using System.Collections.Generic;
+namespace MyCouch
+{
+    public interface IShowParameters
```
Comment (danielwertheim) [inline source/projects/MyCouch.Net45/IShowParameters.cs]: Missing comments for API on properties.

File: source/projects/MyCouch.Net45/Contexts/Documents.cs [hunk 5]
```diff
@@ -177,6 +183,16 @@ public virtual Task<DocumentHeaderResponse> DeleteAsync(string id, string rev)
             }
         }
 
+        public virtual async Task<RawResponse> GetRawAsync(QueryShowRequest request)
```
Comment (danielwertheim) [inline source/projects/MyCouch.Net45/Contexts/Documents.cs]: Would it make more sense to name it `ShowAsync`? I  know that it kind of violates the current namings, but `GetRawAsync` is a bit misleading. Opinion?

**Comment by danielwertheim** [comment]
Looks nice apart from member not being used. Maybe change to `ShowAsync`. Other than that, I can merge it in when ever you feel it's ready.

Nice work.

**Comment by indranilatcal** [comment]
Will take a look at the unused stuff and make the changes, can change it to
ShowAsync.
Will put comments for all public API before replicating the changes to
other forms of projects (4, PCL).

Thanks for the review.

**Comment by indranilatcal** [comment]
Pushed changes through applying review changes. Have a look and merge if appropriate.

**Pull request merged**